src/OrchardCore.Modules/OrchardCore.Media/Indexing/PdfMediaFileTextProvider.cs

```diff
@@ -0,0 +1,32 @@
+using System.IO;
+using Cysharp.Text;
+using UglyToad.PdfPig;
+
+namespace OrchardCore.Media.Indexing
+{
+    public class PdfMediaFileTextProvider : IMediaFileTextProvider
+    {
+        public bool CanHandle(string path)
+        {
+            return path.EndsWith(".pdf");
+        }
+
+        public string GetText(string path, Stream fileStream)
```

Comment:
I used the approach to return `string` here so later perhaps the same providers can be used to build `Content.ContentItem.FullText` too. However, it also means that we'll pass around potentially large strings.

The alternative though would be to let these providers do the indexing directly (i.e. `context.DocumentIndex.Set()`) so not the whole file content needs to be loaded at once. That wouldn't keep the option open to use them with `Content.ContentItem.FullText` too though.

What do you think, @deanmarcussen?

Reply:
What do you think, @deanmarcussen?

Reply:
I think we should keep the FullText Aspect separated with the FileText Aspect. Here, if we want to search in both we can still do it with Lucene. But for making this cleaner I'd not mix those 2 aspects.

Reply:
If there will be a `FileText` aspect then I guess it would need to receive the string content of the file in the same way then. Doing something with aspects is an option for the future though.

Reply:
Yeah, we don't have a FileText aspect and probably won't need. But at the same time if someone wants to add the FileText content in his FullText index it will kind of duplicate indexing for nothing. Here, if we already analyze the FileText document in the index then we should just use it instead of adding it also in the FullText document. It will make the index grow unecessarly. That's more what I meant. Also, in the Search feature we can specify which index column(s) we want to look into. So, it's just a matter of adding the FileText column to the Search configuration.

Reply:
Also, I would tend to create a completely independent search controller for searching in these PDF files. I generally want to search inside the website page content independently of their attached media files. An example could be a search feature for the Media Gallery that would allow to search inside any files text content.

Reply:
For your first message: Since this is what happens now do I understand correctly that what's now in the code is good then?

For the second one: We could spec out a file search feature for Media Library as a next step, the infrastructure for it will be here then.

Reply:
Yes, all good.